src/OmniSharp/Workers/Refactoring/FixUsings.cs

```diff
@@ -0,0 +1,238 @@
+        {
+            bool processMore = true;
+            var ambiguousNodes = new List<SimpleNameSyntax>();
+            var ambiguous = new List<QuickFix>();
+
+            while (processMore)
+            {
+                bool updated = false;
+
+                var syntaxNode = (await document.GetSyntaxTreeAsync()).GetRoot();
+                var nodes = syntaxNode.DescendantNodes()
+                    .OfType<SimpleNameSyntax>()
+                    .Where(x => semanticModel.GetSymbolInfo(x).Symbol == null && !ambiguousNodes.Contains(x)).ToList();
+
+                foreach (var node in nodes)
+                {
+                    document = workspace.GetDocument(fileName);
+                    semanticModel = await document.GetSemanticModelAsync();
+                    var sourceText = (await document.GetTextAsync());
+                    var diagnostics = semanticModel.GetDiagnostics();
+                    var location = node.Identifier.Span;
+                    var actions = new List<CodeAction>();
+                    var usingOperations = new Dictionary<string, CodeActionOperation>();
+                    var pointDiagnostics = diagnostics.Where(d => d.Location.SourceSpan.Contains(location)).ToImmutableArray();
+
+                    if (pointDiagnostics.Any())
+                    {
+                        var pointdiagfirst = pointDiagnostics.First().Location.SourceSpan;
+                        if (pointDiagnostics.Any(d => d.Location.SourceSpan != pointdiagfirst))
+                        {
+                            continue;
+                        }
+                        var context = new CodeFixContext(document, pointDiagnostics.First().Location.SourceSpan, pointDiagnostics, (a, d) => actions.Add(a), CancellationToken.None);
+                        var providers = codeActionProvider.CodeFixes;
+
+                        foreach (var provider in providers)
+                        {
+                            var register = provider.RegisterCodeFixesAsync(context);
+
+                            foreach (var action in actions)
+                            {
+                                if (action.Title.StartsWith("using"))
+                                {
+                                    var operations = await action.GetOperationsAsync(CancellationToken.None).ConfigureAwait(false);
+                                    if (operations != null)
+                                    {
+                                        foreach (var codeOperation in operations)
+                                        {
+                                            if (codeOperation != null)
+                                            {
+                                                usingOperations.Add(action.Title, codeOperation);
```

Comment:
Have reduced the nesting. There was no need to loop over each action/operation for each provider. Looping over the providers separately.